docs/api/plms.yaml

```diff
@@ -285,6 +285,926 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /racks:
+    get:
+      operationId: getRacks
+      summary: Get Racks
+      description: Get all racks.
+      parameters:
+        - in: query
+          name: isVirtual
+          schema:
+            type: boolean
+          description: Return virtual racks or physical racks.
+      tags:
+        - Racks
+      responses:
+        200:
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Racks'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        408:
+          description: Request Timeout
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        500:
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    put:
+      operationId: updateRack
+      summary: Update Rack
+      description: Update a rack.
+      tags:
+        - Racks
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Rack'
```

Comment:
This can contain Id and fields that will be updated instead of all Fields

Reply:
Rack id would be required for the put method. You can check the "rack" component below.